searx/network/network.py

```diff
@@ -144,15 +148,35 @@ async def log_response(self, response: httpx.Response):
             f'HTTP Request: {request.method} {request.url} "{response_line}"{content_type}'
         )
 
-    def get_client(self, verify=None, max_redirects=None):
+    @staticmethod
+    async def check_tor_proxy(client: httpx.AsyncClient, proxies) -> bool:
+        if proxies in Network._TOR_CHECK_RESULT:
+            return Network._TOR_CHECK_RESULT[proxies]
+
+        result = True
+        # ignore client._transport because it is not used with all://
+        for transport in client._mounts.values():  # pylint: disable=protected-access
```

Comment:
check all mounted transports except the default one: 
* when the configuration `all://: socks5h://127.0.0.1:9050`, the default transport is not used
* there is a loophole with `https://somewhere: socks5h://127.0.0.1:9050` and if the engine(s) send(s) requests to another domain.